crates/sparrow-instructions/src/evaluators/equality.rs

```diff
@@ -30,13 +31,58 @@ impl EvaluatorFactory for NeqEvaluatorFactory {
         create_typed_evaluator!(
             &info.args[0].data_type,
             NumericNeqEvaluator,
+            MapNeqEvaluator,
             BoolNeqEvaluator,
             StringNeqEvaluator,
             info
         )
     }
 }
 
+/// Evaluator for `eq` on maps.
+struct MapEqEvaluator {
```

Comment:
Snuck these in here since they were easy and allowed me to continue to use the `create_typed_evaluator` macro for first/last.

Reply:
Although -- map equality is not yet supported by arrow. https://docs.rs/arrow-ord/44.0.0/src/arrow_ord/comparison.rs.html#1746

Reply:
We'll return an internal error if this happens with a sparrow log noting that this is unsupported.

Reply:
Given we want to use `eq_dyn`, we should be able to use an `EqEvaluator` that works for all types.

Within that, we could detect that we're on a map, and then do equality on the key, value and offsets separately. Alternatively, I'd suggest you add a PR to Arrow to support `eq_dyn` on map (shouldn't be too hard to add) and would let us use it.

Reply:
Added a task to the issue. Planning to merge this as is to unblock a demo of map aggregations then follow up with one of the two solutions.